extend deps command with profile flags

joppevos opened the pull request:
## Description

Extends the local operator when running `dbt deps` with the provides profile flags.
This makes the logic consistent between DAG parsing and task running as referenced below

https://github.com/astronomer/astronomer-cosmos/blob/8e2d5908ce89aa98813af6dfd112239e124bd69a/cosmos/dbt/graph.py#L247-L266

## Related Issue(s)

<!-- If this PR closes an issue, you can use a keyword to auto-close. -->
<!-- i.e. "closes #0000" -->
closes #658 

## Breaking Change?

<!-- If this introduces a breaking change, specify that here. -->

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

tatiana reviewed (commented):
This was super quick, @joppevos , thank you for fixing this issue.

The only thing missing is a test. I believe we could add a unit test in `astronomer-cosmos/tests/operators/test_local.py`, calling `run_command` and mocking `run_subprocess`, and checking if we're passing the expected arguments.

There is something similar in:
https://github.com/astronomer/astronomer-cosmos/blob/188fe56994e29037eda1777974faa45a183ebc1f/tests/operators/test_local.py#L369-L373

The motivation to have a test is to make sure we don't re-introduce this bug in the future.

joppevos commented on cosmos/operators/local.py:
```diff
@@ -225,6 +219,18 @@ def run_command(
                     self.profile_config.target_name,
                 ]
 
+                if self.install_deps:
```
@tatiana We could separate the command creation outside of `run_command`  and bring it into `build_and_run`, then it's easier to test for the arguments

tatiana replied:
It sounds like a good plan, @joppevos !

joppevos commented:
> Thanks for working on this and addressing the feedback so quickly, @joppevos ! We'll merge it once the checks pass.

Sounds good! Thanks for the support @tatiana :)

tatiana commented:
Thanks for the contribution, @joppevos  , we released it as part of Cosmos 1.2.4 yesterday:
https://github.com/astronomer/astronomer-cosmos/releases/tag/astronomer-cosmos-v1.2.4
